ide/che-core-ide-ui/src/main/resources/org/eclipse/che/ide/ui/smartTree/TreeStyles.css

```diff
@@ -70,6 +70,11 @@
 
 .selected {
     background-color: projectExplorerSelectedRowBackground;
+    color: projectExplorerJointContainerFill !important;
+}
+
+.selected * {
+    color: projectExplorerJointContainerFill !important;
```

Comment:
Add comment to this block of code, that describe hack, that selected node text forces to apply another color